pkg/controller/migmigration/task.go

```diff
@@ -95,6 +97,8 @@ var StageItinerary = Itinerary{
 	Steps: []Step{
 		{phase: Created},
 		{phase: Started},
+		{phase: Refresh},
+		{phase: EnsureRefreshed},
```

Comment:
while it is good to ensure refresh at the start of the Itinerary, would it probably make more sense to refresh before every task is run? 

It's just the refreshing here that does not have much say if a lot of time is spent between this phase and the velero backup and restore phases.

Reply:
Interesting thought. The effect of that would be that every MigMigration reconcile that runs a migration phase would also enqueue reconciles on the MigPlan, MigClusters, and MigStorage. 

I think this could actually be useful, although I hesitate to say we should **wait** on the refresh to finish successfully before continuing like we do at the beginning. I'll talk it over with @jortel as well and get his thoughts.

Reply:
I agree that there will be some time gap between starting the migration and the initial backup, but I am aware of any common scenarios where we really need MigCluster and MigStorage reconcile to run again afterwards. The BSLs and VSLs set up should still be around and it seems unlikely they became invalid if they were OK a few minutes ago. The MigCluster connection is being monitored by a ProviderSource watch mechanism on an ongoing basis. 

```
	// Watch for changes to cloud providers.
	err = c.Watch(
		&ProviderSource{
			Client:   mgr.GetClient(),
			Interval: time.Second * 30},
		&handler.EnqueueRequestForObject{})
```

From this discussion I'm realizing a flaw in this PR as it stands today: a refresh on one MigPlan will eventually trigger reconciles on all other MigPlans connected to the same MigStorage or MigClusters.  I'm trying to think of ways to overcome this.

Reply:
Idea that came to me on this for folks to shoot holes in: what if we _only_ run heavy MigPlan processes (PV discovery, GVK diff) when a refresh is  requested? Otherwise we do a lightweight reconcile that just runs validations quickly.

Reply:
Idea from Jeff: since MigPlan already watches MigMigrations, we can modify the MigPlan watch predicate for MigMigrations to allow reconcile events to come thorough every time the phase changes on MigMigration.

https://github.com/konveyor/mig-controller/blob/c9218dd717b295f8f2e817ed466d0576924606d1/pkg/controller/migplan/predicate.go#L164-L190

Reply:
Probably want to stop skipping storage validation while suspended https://github.com/konveyor/mig-controller/blob/6c28e299c8086ed7754ab6c2d8a760aecefd93f7/pkg/controller/migplan/storage.go#L21-L24

Reply:
@alaypatel07 now thinking we will pull the StorageClass info out of MigCluster and into MigPlan PV discovery routine so that we don't have to refresh MigCluster and can avoid having as much refresh code.